### PR TITLE
Add default docstrings to OSPlugin subclass methods

### PR DESCRIPTION
**Pull request by pyrco**

### CODE REVIEW

File: dissect/target/plugin.py [hunk 1]
```diff
@@ -292,18 +292,42 @@ class OSPlugin(Plugin):
...
+            docstring = method.__doc__
+
+            if method and not docstring:
+                if hasattr(method, "__func__"):
+                    method = method.__func__
+                method.__doc__ = os_docstring
+
     def check_compatible(self) -> bool:
-        """OSPlugin's use a different compatibility check, override the default one."""
+        """OSPlugin's use a different compatibility check, override the one from the Plugin class.
```
Comment (Schamper) [inline dissect/target/plugin.py:321]:
```suggestion
        """OSPlugin's use a different compatibility check, override the one from the :class:`Plugin` class.
```
I don't quite remember the syntax, but I think it was like this?

File: dissect/target/plugin.py [hunk 2]
```diff
@@ -292,18 +292,42 @@ class OSPlugin(Plugin):
...
+
+        Returns:
+            This function always returns ``True``."""
```
Comment (Schamper) [inline dissect/target/plugin.py:324]:
```suggestion
            This function always returns ``True``.
        """
```

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugin.py [hunk 3]
```diff
@@ -292,18 +292,43 @@ class OSPlugin(Plugin):
     This provides a base class for certain common functions of OS's, which each OS plugin has to implement separately.
 
     For example, it provides an interface for retrieving the hostname and users of a target.
+
+    All derived classes MUST implement ALL the classmethods and exported
+    methods with the same @classmethod or @export(...) annotation.
```
Comment (Schamper) [inline dissect/target/plugin.py:297]:
```suggestion
    methods with the same ``@classmethod`` or ``@export(...)`` annotation.
```

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugin.py (hunk 3 above)
Comment (pyrco) [reply dissect/target/plugin.py]: Changed a few more

**Review by Schamper — APPROVED** [review]

**Pull request merged**